synapse/replication/tcp/handler.py

```diff
@@ -404,3 +418,52 @@ def stream_update(self, stream_name: str, token: str, data: Any):
         We need to check if the client is interested in the stream or not
         """
         self.send_command(RdataCommand(stream_name, token, data))
+
+
+UpdateToken = TypeVar("UpdateToken")
+UpdateRow = TypeVar("UpdateRow")
+
+
+def _batch_updates(
```

Comment:
Why do we need a new impl for this? Can't we share the existing one?

Reply:
well, it batches them in a slightly different way. I agree that it would be nice not to have two different impls, but I couldn't easily think of a way to share the code.

Reply:
Oh yes, of course it needs to be different.